pkg/apis/sources/v1alpha1/kafka_types.go

```diff
@@ -121,6 +127,11 @@ type KafkaSourceStatus struct {
 	// * SinkURI - the current active sink URI that has been configured for the
 	//   Source.
 	duckv1.SourceStatus `json:",inline"`
+
+	// Total number of non-terminated replicas targeted by this source
+	// +optional
+	// For round-tripping only
+	Replicas int32 `json:"replicas,omitempty"`
```

Comment:
Is that causing some "migration" problems?

I know we do lack "upgrade tests" for the kafka stuff

Reply:
I don't think so since this is a new field

Reply:
Does it make a sense to move this to `duckv1.SourceStatus`?

Reply:
I think so, as an optional field. From a conformance PoV, a source *may* support horizontal scaling. The counter argument is we tend to only declare fields that are used. Not all sources supports `replicas`. 

If we decide to add it to `duckv1.SourceStatus` later on, it won't be breaking the API so I suggest we leave it this way for now.